### PR TITLE
Mohammed.log stat

### PR DESCRIPTION
**Pull request by MohammedAlewi**

### CODE REVIEW

File: Backend/controllers/PatientsController.js [hunk 1]
```diff
@@ -1,6 +1,97 @@
...
+    } else {
+        filter = {_id: null} //returns null on User
+    }
+    let user = await User.findOne(filter);
+    if(!user){
+        let unusedUsername = `${patient.first_name}-${patient.last_name}-${Date.now().toString().substring(6, 13)}`;
+        const foundUser = await User.findOne({username: unusedUsername});
+        while (foundUser){
+            unusedUsername = `${patient.first_name}-${patient.last_name}-${Date.now().toString().substring(6, 13)}`;
+            const foundUser = await User.findOne({username: unusedUsername});
```
Comment (Infinidrix) [inline Backend/controllers/PatientsController.js:26]: I think we need to remove const here

File: Backend/controllers/TestReportController.js [hunk 2]
```diff
@@ -3,6 +3,52 @@ var mongoose = require("mongoose");
...
+    let user = await User.findById(report.user_id);
+    
+    if(user.patient_info){
+        let patient = await Patient.findById(user.patient_info)
+
+        let prev_date = new Date(patient.updated_at)
+        prev_date.setHours(0,0,0,0);
+
+        if(prev_date == date){
+            let log =await PatientLog.findOne({date:prev_date,test_status:patient.status});
```
Comment (Infinidrix) [inline Backend/controllers/TestReportController.js:25]: Can't we do a findOneAndUpdate

**Review by Infinidrix — COMMENTED** [review]
Small comments

**Review by Khalid-Sultan — APPROVED** [review]

**Review by Infinidrix — APPROVED** [review]
Looks good to me :1st_place_medal:

**Pull request merged**